### ISSUE TITLE
Groups and stacked/dodged positions.

### ISSUE THREAD

**Issue by AlFontal** [has2k1/plotnine#608]
I'm dealing with a figure where I would like to be able to dodge columns based on the `group` aesthetic while having the columns stay stacked on the `fill` aesthetic. 

My intuition was that for geometries that use the `group` aesthetic, color/fill is assumed to be used as the `group` when it is not explicitly mapped to a variable. This seems to be the case with `geom_line`, for instance. 

However, when using `position_dodge` and an explicitly mapped `group`, the bars appear to still be dodged by the `group + fill` variable. 

To show some examples with a toy dataset:

```python
df = pd.DataFrame(dict(
    day=np.repeat(range(1, 5), 3),
    species=np.tile(['A', 'B', 'C'], 4),
    counts=np.random.randint(0, 200, 12),
    replicate=1
))

df_replicates = df.query('day==2').assign(replicate=2).assign(counts=np.random.randint(0, 200, 3))

full_df = pd.concat([df, df_replicates])
```

|   day | species   |   counts |   replicate |
|------:|:----------|---------:|------------:|
|     1 | A         |       49 |           1 |
|     1 | B         |       92 |           1 |
|     1 | C         |      195 |           1 |
|     2 | A         |      110 |           1 |
|     2 | B         |      150 |           1 |
|     2 | C         |       32 |           1 |
|     3 | A         |        9 |           1 |
|     3 | B         |      146 |           1 |
|     3 | C         |      195 |           1 |
|     4 | A         |      161 |           1 |
|     4 | B         |      190 |           1 |
|     4 | C         |      100 |           1 |
|     2 | A         |       88 |           2 |
|     2 | B         |       87 |           2 |
|     2 | C         |       32 |           2 |

In this case, we have two different samples/replicates in day 2.
## Default Behaviour

```python
(p9.ggplot(full_df)
 + p9.aes('day', 'counts', fill='species', group='replicate')
 + p9.geom_col()
)
```
![image](https://user-images.githubusercontent.com/10958332/183124367-2ee8958f-3fe5-4249-9faa-cd5dfb99ed59.png)

Since the default position is `stack`, specifying the `group` as the replicate stacks the bars for each of the replicates of day 2 following the order of species-replicate:  `A-1, B-1, C-1, A-2, B-2, C-2`. Fair enough, this is probably the expected and desired behaviour.

## Dodged Behaviour

The issue arises when we attempt to dodge the groups, so that the height of the bars still represents the total counts in a single sample:

### `position_dodge`

```
(p9.ggplot(full_df)
 + p9.aes('day', 'counts', fill='species', group='replicate')
 + p9.geom_col(position='dodge')
)
```

![image](https://user-images.githubusercontent.com/10958332/183125361-4aa0beb0-e908-44d3-971f-0c1d048e64f4.png)

The `group` (replicates) have now been properly dodged/unstacked in day 2, but the previously stacked species all start now in the same x-y coords of the day-group instead of being stacked. Lowering the `alpha` a tad helps to see it better:

![image](https://user-images.githubusercontent.com/10958332/183125920-eee074d9-254f-4774-a603-44165def2ae3.png)

### `position_dodge2`

The results with `position_dodge2` are also problematic, since the width of the bars doesn't seem to be properly calculated (day 1 values become half as wide, while values for day 2 stay the same and overlap with values from the other days). 

In any case, this completely unstacks the species/fill aesthetic, which is something we don't want even if bars' width was properly calculated.

## The hacky solution

I thought that manually changing the `x` values (works in this case but would be a bit harder if the `x` variable was a Categorical, for instance) and generating a `width` variable to pass as an aesthetic manually setting the width of the day 2 values to .25 should allow me to make this work:

```python
(full_df
 .assign(day=lambda dd: np.where(dd.replicate == 2, dd.day + .22, dd.day))
 .assign(day=lambda dd: np.where(dd.day==2, dd.day - .22, dd.day))
 .assign(width=lambda dd: np.where(dd.day.round() == 2, .4, .8))
 .pipe(lambda dd: p9.ggplot(dd)
 + p9.aes('day', 'counts', fill='species', group='replicate')
 + p9.geom_col(position='stack', mapping=p9.aes(width='width'))
)
```
![image](https://user-images.githubusercontent.com/10958332/183128833-796ced4e-5ed7-47f8-bbb9-61b4ac885a63.png)

I actually managed to solve this while writing the issue with this hacky solution, but still, it feels like there might be some way to include this behaviour in some `position` variety? I don't know if this situation is common enough to justify generating a specific `position_dodgestack` or something like that, but what do you think?

**Comment by has2k1** [comment]
Yes this is a `dodgestack` or `stackdodge` situation. Though I do not think it would be common enough to warrant an implementation. Plus the resulting graphic has missing information! There is no guide for the two different split bars at location 2. You need to know about the replication to understand what is shown!
